tensorflow_model_optimization/python/core/clustering/keras/clustering_registry.py

```diff
@@ -27,21 +27,22 @@ class ConvolutionalWeightsCA(AbstractClusteringAlgorithm):
   """Look-ups for convolutional kernels, e.g. tensors with shape [B,W,H,C]."""
 
   def get_pulling_indices(self, weight):
+    wt_dim = len(weight.shape)
     clst_num = self.cluster_centroids.shape[0]
-    tiled_weights = tf.tile(tf.expand_dims(weight, 4), [1, 1, 1, 1, clst_num])
+    tiled_weights = tf.tile(tf.expand_dims(weight, wt_dim), [1 for _ in range(wt_dim)] + [clst_num])
 
     # Do the ugly reshape to the clustering points
     tiled_cluster_centroids = tf.stack(
         [tf.tile(tf.stack(
-            [tf.reshape(self.cluster_centroids, [1, 1, clst_num])] *
-            weight.shape[-2], axis=2),
-                 [weight.shape[0], weight.shape[1], 1, 1])] * weight.shape[-1],
-        axis=3)
+            [tf.reshape(self.cluster_centroids, [1 for _ in range(wt_dim-2)] + [clst_num])] *
+            weight.shape[-2], axis=wt_dim-2),
+                 [i for i in weight.shape[:-2]] + [1, 1])] * weight.shape[-1],
+        axis=wt_dim-1)
```

Comment:
Seems a little clunky with all the list comprehensions, is it possible to make this a bit more readable?